src/fast_sincos.c

```diff
@@ -0,0 +1,407 @@
+            return (((
+                -1.9462116998273101e-4
+                * angleRadiansSquared +8.3315846064878458e-3)
+                * angleRadiansSquared -1.6666636754299513e-1)
+                * angleRadiansSquared +9.9999998617934201e-1)
+                * angleRadians;
+    }
+}
+
+/**
+ * The Chebyshev approximation is used to compute the cosine of the angle.
+ * Coefficients were obtained manually with the Remez algorithm.
+ * Horner's method (nested multiplication) is then used for faster calculations.
+ * The range for cos is [-PI / 4, PI / 4].
+ * This range was chosen as some of the coefficients become negligible,
+ * reducing the total number of multiplications.
+ * @param angleRadians The angle, in radians, within the range [-PI / 4, PI / 4].
+ * @param sinDegree The degree of the Chebyshev approximation for sin.
+ * Because this approximation is for cos, the actual degree of the approximation
+ * will be "sinDegree + 1".
+ * Higher is more accurate, but slower.
+ * @return A cosine approximation of the angle.
+*/
+static fast_sincos_real fastCosChebyshev(const fast_sincos_real angleRadians, const int sinDegree) {
+    fast_sincos_real angleRadiansSquared = angleRadians * angleRadians;
+    switch (sinDegree) {
+        case 3: // Actually computes for degree 4
+            return (
+                4.0398535966168857e-2
+                *angleRadiansSquared-4.9970814035466399e-1)
+                *angleRadiansSquared+9.9999003495519596e-1;
+        case 5: // Actually computes for degree 6
+            return ((
+                -1.3585908510113299e-3
+                *angleRadiansSquared+4.1655026884251524e-2)
+                *angleRadiansSquared-4.9999856695848848e-1)
+                *angleRadiansSquared+9.9999997242332292e-1;
+        default:
+        case 7: // Actually computes for degree 8
+            return (((
+                2.4379929375956876e-5
+                *angleRadiansSquared-1.3886619210252882e-3)
+                *angleRadiansSquared+4.1666616739207635e-2)
+                *angleRadiansSquared-4.9999999615433476e-1)
+                *angleRadiansSquared+9.9999999995260044e-1;
+        
+    }
+}
+
+/**
+ * @brief Fast sine computation with lookup table, using interpolation
+ * @param angleRadians The angle, in radians.
+ * @return A sine approximation of the angle.
+*/
+static fast_sincos_real lookupSinInterpolate(const fast_sincos_real angleRadians) {
+  
+  fast_sincos_real scaledAngle = angleRadians * LOOKUP_SCALE_FACTOR;
+  int negativeFactor = 0;
+  if (scaledAngle < 0) {
+    negativeFactor = 1;
+    scaledAngle = -scaledAngle;
+  }
+
+  long roundedAngle = scaledAngle;
+  // Get the remainder in fixed point, for faster calculations
+  // The function below allows for fast floating point multiplication by power of 2
+  uint16_t remainder = ldexpf(scaledAngle - roundedAngle, LOOKUP_REMAINDER_BITS); 
+  uint16_t index = roundedAngle & LOOKUP_INDEX_MASK;  // This & replaces modulus
```

Comment:
A modulus operation is implemented using a logical and. 
From what I can see it would be more readable to directly execute the modulo and remove this comment. What do you think ?

Reply:
Some "in-house" compilers might not make this optimisation. For the sake of understanding why it provides a significant acceleration, I think it is better to leave this `&` here.

Reply:
This is not an optimization, this is literally how a modulo operation is implemented in assembly.